server/app/services/program/ProgramService.java

```diff
@@ -1403,8 +1421,10 @@ public CompletionStage<ImmutableList<ProgramDefinition>> syncQuestionsToProgramD
     for (ProgramDefinition programDef : programDefinitions) {
       ProgramModel p = programDef.toProgram();
       p.refresh();
-      // We only need to get the question data if the program has eligibility conditions.
-      if (programDef.hasEligibilityEnabled()) {
+      // We only need to get the question data if the program has eligibility conditions and the
```

Comment:
question (non-blocking): I have a question mostly for my own understanding. I know that if an applicant starts applying to a program, then an admin makes changes to the program and publishes a new version, the applicant should still see the old version. Each time an admin makes a new version of a program, it gets a new program ID right? So the cache should still work correctly for applicants filling out old versions of programs because the ID will still be tied to the old version of the program definition in the cache?

Reply:
Yeah that is all correct! Once a program is published, there will be no changes to that program (when looked up by ID) and when that program is edited, it will keep the name and get a new ID. The only time a program can change is if its in draft mode, which is why we only cache programs that are active or obsolete.

Reply:
Excellent, thanks for the clarification!